units/unit.cpp

```diff
@@ -3,79 +3,47 @@
 /// @file
 /// @author Simon Heybrock
 /// @author Neil Vaytet
+#include <regex>
 #include <stdexcept>
-#include <vector>
 
-#include <boost/lexical_cast.hpp>
-#include <boost/units/cmath.hpp>
-#include <boost/units/io.hpp>
+#include <units/units.hpp>
 
-#include "scipp/units/boost_units_util.h"
 #include "scipp/units/except.h"
 #include "scipp/units/unit.h"
 
 namespace scipp::units {
 
-// Helper to check whether type is a member of a given std::tuple
-template <typename T, typename VARIANT_T> struct isTupleMember;
-template <typename T, typename... ALL_T>
-struct isTupleMember<T, std::tuple<ALL_T...>>
-    : public std::disjunction<std::is_same<T, ALL_T>...> {};
-// Helper to make checking for allowed units more compact
-template <class Units, class T> constexpr bool isKnownUnit(const T &) {
-  return isTupleMember<T, Units>::value;
-}
-
-template <class... Ts> auto make_name_lut(std::tuple<Ts...>) {
-  return std::array{std::string(boost::lexical_cast<std::string>(Ts{}))...};
+Unit::Unit(const std::string &unit)
+    : Unit(llnl::units::unit_from_string(unit == "dimensionless" ? "" : unit)) {
 }
 
 std::string Unit::name() const {
-  static const auto names = make_name_lut(supported_units_t{});
-  return names[index()];
+  auto repr = to_string(m_unit);
+  // Replace 'LATIN CAPITAL LETTER A WITH RING ABOVE' by 'ANGSTROM SIGN'
+  repr = std::regex_replace(repr, std::regex("Å"), "Å");
```

Comment:
Search and replacement terms the same?

Reply:
I thought it might be that. Should have gone to raw view.

Reply:
But both are `'ANGSTROM SIGN'`, no?

Reply:
No, see comment in the line above. The replaced one is https://www.compart.com/en/unicode/U+00C5.

Reply:
Ah, ok